同步上游更新

pityonline opened the pull request:
@Donng 看一下这个 PR 吧，请关注以下几点：

1. 只会取到上游更新，不会把我们这个仓库的历史提交重新合并一次。
2. 只合并上游更新，不会覆盖或移除 zh-CN 目录。
3. 我在本地测试 `./build.sh` 是通过的。
4. .travis.yml 有些变化，应该不影响中文翻译在线网页的生成，CI 也应该是通过的。

我之前的做法是在本地 merge 好上游后直接 push 到这里的 master，现在看来网页发 PR 效果应该是一样的。cc @polaris1119 @flw-cn

mukulrawat1986 committed:
Convert int to Bitcoin type.

mukulrawat1986 committed:
Merge remote-tracking branch 'upstream/master'

Merging with upstream

spacewarped committed:
Fix name of tests for recording wins of new players

pityonline committed:
fix markdown format

awulkan committed:
Fix typos and other small things

Tell me if there's something you want me to change back.

awulkan committed:
Add missing word

quii committed:
Merge pull request #126 from chriskdon/bugfix/io-test-names

Fix name of tests for recording wins of new players

quii committed:
Merge pull request #127 from AWulkan/patch-3

Fix typos and other small things

quii committed:
Merge pull request #128 from AWulkan/patch-4

Add missing word

pityonline committed:
fix grammar on command-line

quii committed:
Merge pull request #129 from mukulrawat1986/master

Convert int to Bitcoin type.

quii committed:
script to build epub

quii committed:
wip - build epub in github with a view to automating it as a release

quii committed:
lets try the deploy

quii committed:
typo

quii committed:
not the highest hopes of this working

quii committed:
another go for pandoc with docker

quii committed:
probably a bit overkill trying to build the epub twice

quii committed:
prevent generated epub being destroyed

quii committed:
omg

quii committed:
Another stab at releasing the epub

quii committed:
Only do release on tags

pityonline committed:
trivial grammar and style tweak on reflection.md

pityonline committed:
move docker pull action into specific stage

pityonline committed:
format .travi.yml

quii committed:
Merge pull request #130 from pityonline/fix/reflection

trivial grammar and style tweak on reflection.md

quii committed:
Merge pull request #131 from pityonline/chore/ci-stage

shorten CI build time, format .travis.yml

quii committed:
Merge pull request #133 from pityonline/fix/commond-line

fix grammar and style on commond-line.md

mukulrawat1986 committed:
Name the function which holds our subtest.

mukulrawat1986 committed:
Merge branch 'master' of https://github.com/quii/learn-go-with-tests

Merge

quii committed:
clarification on what a package is

quii committed:
wip - starting websockets chapter

quii committed:
starting to write the codes for serving websocket page

quii committed:
fmt

quii committed:
more notes

quii committed:
more work on websockets and start of todo thing

mukulrawat1986 committed:
Merge branch 'master' of https://github.com/quii/learn-go-with-tests

quii committed:
wip - start on a why chapter to introduce why tdd is important

quii committed:
more on why

quii committed:
some more formatting and summary for 'why'

quii committed:
more on why

quii committed:
more tinkering with why

quii committed:
why - some formatting

quii committed:
messed up with vim somehow

quii committed:
why - more corrections

quii committed:
formatting of why

quii committed:
and yet more corrections

quii committed:
corrections

quii committed:
more work on the WebSockets

HaraldNordgren committed:
Bump Go versions

quii committed:
Merge pull request #135 from HaraldNordgren/go_versions

Bump Go versions

pdbrito committed:
Rename `FileSystemStore` to `FileSystemPlayerStore`

This is how it's names in later parts of the document as well as in
all the code examples

quii committed:
Merge pull request #136 from pdbrito/fix-struct-name

Rename `FileSystemStore` to `FileSystemPlayerStore`

quii committed:
Merge pull request #134 from mukulrawat1986/master

Name the function which holds our subtest.

quii committed:
vendoring deps to fix the build in travis hopefully

pdbrito committed:
Update database to refactored type

Also align struct field values

pdbrito committed:
Add some whitespace around ==

quii committed:
Merge pull request #137 from pdbrito/fix-typo-in-io-code-examples

Fix typo in io code examples

danillouz committed:
Fix typos, grammar and add context info

- Add the word "have".
- Remove "and get rid of the `given` piece", since it's not clear what `given` refers to.
- Add more context to Reference Types. It's not clear what is meant by "use". Now it's clear you can read from a nil map, but writes to a nil map cause a panic.
- Remove `**` from `Add**`, since it's inconsistent and not clear what it means.

quii committed:
Merge pull request #138 from danillouz/patch-1

Fix typos, grammar and add context info

jdsutherland committed:
Fix syntax error

quii committed:
Merge pull request #139 from jdsutherland/fix-json-syntax-error

Fix formatting

quii committed:
add note about epub

jdsutherland committed:
Fix erroneous code reference

It seems that `ReadString` should be `readLine` as there is no current
reference to `ReadString` in the code.

jdsutherland committed:
Replace code reference with actual needed change

The text says to update the code storing a game variable but we actually
need to call `PlayPoker()`. It seems the text would make more sense
listing the code needed to run the game.

quii committed:
Merge pull request #142 from jdsutherland/time.md-updating-main-L411

time.md - updating main code reference

quii committed:
Merge pull request #141 from jdsutherland/time.md-ReadString-errata

time.md errata - ReadString invalid code reference

quii committed:
this chapter needs some more work before it can be considered done

quii committed:
some corrections on command line, history went a bit weird during some edits

quii committed:
some corrections and edits around time

quii committed:
q and a - os-exec

quii committed:
spacing

quii committed:
bring naming inline

quii committed:
little edits

quii committed:
dont need all that

quii committed:
adding q and a section

quii committed:
link to code

quii committed:
damn spaces

quii committed:
fix package name

quii committed:
fixing Keith's spelling

quii committed:
fixing summary.md

quii committed:
Bit more waffle on sep of concerns

quii committed:
twiddling

quii committed:
some boldy shouty stuff

quii committed:
and more furniture arranging

Omar-Khawaja committed:
fixing title capitalization

Omar-Khawaja committed:
add links to contribute

quii committed:
Merge pull request #146 from Omar-Khawaja/add-github-link

add links to contribute

quii committed:
Merge pull request #145 from Omar-Khawaja/edit-title-capitalization

fixing title capitalization

Omar-Khawaja committed:
adding a link for the Github repo so it's immediately clear where to go from website to contribute

quii committed:
Merge pull request #147 from Omar-Khawaja/edit-gb-readme

adding a link for the Github repo in gb-readme

quii committed:
small edits to websockets

Commit:
add link to prev chapter

add link to section for easy navigation

quii committed:
Merge pull request #148 from guylyons/patch-1

add link to prev chapter

darjun committed:
fix a typo - should use slice instead of array

quii committed:
Merge pull request #150 from darjun/master

fix a typo - should use slice instead of array

quii committed:
big picture time

quii committed:
pictures

quii committed:
formatting fun

quii committed:
center it

quii committed:
Update gb-readme.md

quii committed:
refactoring toward game being useable within context of web server

quii committed:
spelling

quii committed:
more progress on the web sockets

quii committed:
showing blind alerts working

quii committed:
you're soooo naive

quii committed:
wip - error types

quii committed:
more work on error types

quii committed:
some blurb

quii committed:
breaking it up with titles a bit

quii committed:
more blurb

quii committed:
and more corrections

quii committed:
trying to clarify some points

quii committed:
tl;dr for the error types

quii committed:
no longer wip on error types

quii committed:
more furniture arranging

quii committed:
Update error-types.md

quii committed:
websockets - wip - found a nice way to test that blinds get written, just need to write it up

quii committed:
wip - websockets testing

quii committed:
some corrections to websockets and an attempt at a cover image for the epub

quii committed:
add why to epub

quii committed:
typo

quii committed:
tag 0.2.0

quii committed:
update cover

quii committed:
git fail

quii committed:
fixes 115

quii committed:
be more explicit around 'murky' section in time.md which hopefully fixes #143

quii committed:
spelling

quii committed:
make the sad spelling time stop

quii committed:
more websockets progress

quii committed:
lil refactor

quii committed:
and another refactor

quii committed:
websockets - remove nasty sleep

quii committed:
websockets - explain how to remove nasty sleep

quii committed:
websockets - fix some typos and a draft of wrapping up

quii committed:
websockets - no longer WIP (perfect is enemy of good)

quii committed:
new cover improvements thanks @gypsydave5

quii committed:
changed my mind, added the words back to cover

quii committed:
locks - first iteration

quii committed:
more lock wip

quii committed:
sync it

quii committed:
typo

quii committed:
spelling

quii committed:
wrapping up sync

quii committed:
some editing of sync

quii committed:
ensure mutexes arent copied

quii committed:
corrections to sync and no longer WIP

quii committed:
output as PDF too

quii committed:
updates on community convention around mutex naming and position plus clarification of release in travis

quii committed:
emphasise the dangers of embedding

quii committed:
explain why embedding with mutexes bad a little more

quii committed:
i cannot spell

Commit:
replaced englishPrefix with helloPrefix in last refactoring example, since englishPrefix does not exist.

Commit:
replaced all instances of helloPrefix with englishPrefix

Commit:
fixed a small typo under the benchmarks section

quii committed:
Merge pull request #151 from robertguss/master

replaced englishPrefix with helloPrefix in last refactoring example

quii committed:
tidying up readme

quii committed:
less silly tidying..

quii committed:
context - wip, trying out some ideas to demonstrate context

tracypholmes committed:
Update to correct ln 358 error msg

Switched "hola" and "hello"

quii committed:
Merge pull request #152 from tracypholmes/patch-1

Update to correct ln 358 error msg

quii committed:
use correct context package

quii committed:
context - wip - some writing on first few iterations

quii committed:
context - wip - some rearranging

quii committed:
context - wip - refactoring

quii committed:
context - wip - moving forward

quii committed:
context - some notes on values

quii committed:
make it clear this bit isnt done yet

quii committed:
context - wip - refactor to pass context through call stack

quii committed:
context - dont need to spy on context anymore and fixed linting issues

quii committed:
context - snappier opening text

quii committed:
context - more comments on context.value

quii committed:
add context to listings

sleeping-barber committed:
Remove wrong header

When looking at the generated PDF I recognized, that the `golang challenge` quote had a hashtag which will be interpreted as a Header and so the ToC was wrong.

quii committed:
Merge pull request #153 from midnightrun/patch-1

Remove wrong header

quii committed:
context - illustrate cancellation better

quii committed:
spelling

quii committed:
remove uber leak thing

quii committed:
some improvements for PDF

quii committed:
fixes #157 - close test file before removing

quii committed:
small edits to pointers

glauco committed:
Updates Homebrew installation instructions

The `--cross-compile-common` flag is no longer supported. It was removed by the following commit:
https://github.com/Homebrew/homebrew-core/commit/bd56fd444ce3d21b647558d8740f6aade9180cba

glauco committed:
Updates VS Code installation instructions

You no longer need to add a tap in order to use homebrew casks.

glauco committed:
Updates command to use $GOPATH

We can avoid a mistake by using `$GOPATH` instead of `~/go`, in case reader has set its path to be something else.

quii committed:
Merge pull request #159 from glauco/patch-1

Updates Go installation instructions

quii committed:
Merge pull request #160 from glauco/patch-2

Updates VS Code installation instructions

quii committed:
Merge pull request #161 from glauco/patch-3

Updates mkdir to use $GOPATH

tracypholmes committed:
ln 475 - Added space before brackets

quii committed:
Merge pull request #162 from tracypholmes/patch-2

ln 475 - Added space before brackets

jossy committed:
Replace Go Meta Linter with GolangCI-Lint

quii committed:
Merge pull request #163 from jossy/master

Replace deprecated Go Meta Linter with GolangCI-Lint

abhishek-jaisingh committed:
Fix Broken Link

Fix the link to Scala - Just enough rope to hang yourself by prepending www to the url

quii committed:
Merge pull request #164 from abhishekjiitr/patch-1

Fix Broken Link

tracypholmes committed:
moved source control verbiage

quii committed:
Merge pull request #165 from tracypholmes/master

moved source control verbiage

jossy committed:
Make it clear when names for test cases are introduced

jossy committed:
Explain the renaming of "want" to "hasArea"

jossy committed:
Fix some tiny grammatical errors

jossy committed:
Make format string explanation a bit more readable

quii committed:
Merge pull request #167 from jossy/master

A few small updates for the last "output" section of " Structs, methods & interfaces"

panakour committed:
fixes typo

quii committed:
Merge pull request #168 from panakour/master

fixes typo

abhishek-jaisingh committed:
Fix Failing Test due to Imprecise value of Pi

quii committed:
Merge pull request #169 from abhishekjiitr/fix-struct-pi-val

Fix Failing Test due to imprecise value of Pi

Commit:
Update install-go.md

gypsydave5 committed:
Fixing the Manny Lehman link

quii committed:
Merge pull request #175 from gypsydave5/patch-9

Fixing the Manny Lehman link

quii committed:
Merge pull request #170 from keramiozsoy/debugger-link

Debugger link can't go directly in install-go.md

quii committed:
attempt at addressing #173

quii committed:
Update arrays-and-slices.md

quii committed:
Update arrays-and-slices.md

kandros committed:
fix after/before in mocking

quii committed:
Merge pull request #178 from kandros/fix-mocking-before-after

fix after/before in mocking

amnay-mo committed:
close FileSystemStore file

quii committed:
Merge pull request #179 from amnay-mo/master

close FileSystemStore file

kandros committed:
fix typo ✍

quii committed:
Merge pull request #181 from kandros/master

fix typo ✍

kandros committed:
fix package name consistency

kandros committed:
follow consistency of separated test file

kandros committed:
sync code and md

vpkopylov committed:
Fix english hello prefix const

quii committed:
Merge pull request #182 from kandros/context-fix

Context fix

quii committed:
Merge pull request #183 from vpkopylov/fix-eng-hello-prefix

Fix english hello prefix const

vinhnx committed:
Fix GitHub link to `filebuffer` library author

quii committed:
Merge pull request #186 from vinhnx/patch-1

Fix GitHub link to `filebuffer` library author

flw-cn commented:
更新内容好多呀。
我看 Travis CI 出错了，没仔细看，有一些依赖代码有问题？

donng commented:
不重新提交应该是更好一些，因为我执行的 rebase 会自动把本地的 commit 提到前面。
测试没有通过是因为 travis 配置文件也被同步了，我改成原来的就能通过，具体是什么原因就不清楚了。

pityonline committed:
fix CI for forks

As an upstream, this repo depends on itself, but for a fork, it depends
on this repo. In Travis CI, `install: true` skipps `go get` for Go
projects, so the forks won't get the dependence and will cause the
CI failure. This is fixed by installing dependence in `before_install`.

Ref: https://docs.travis-ci.com/user/job-lifecycle/#skipping-the-installation-phase

quii committed:
Merge pull request #187 from pityonline/master

fix CI for forks

pityonline commented:
@Donng 看起来了正常了，再帮忙 review 看看是否还有别的问题。
